### PR TITLE
[python] Move temporary feature flag into a module

### PR DESCRIPTION
**Pull request by johnkerl**
**Issue and/or context:** Follow-up from here: https://github.com/single-cell-data/TileDB-SOMA/pull/3001#issuecomment-2355914110

**Changes:**

**Notes for Reviewer:**

### CODE REVIEW

File: apis/python/src/tiledbsoma/_flags.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+_new_shape_feature_flag = os.getenv("SOMA_PY_NEW_SHAPE") is not None
+
+
+def _new_shape_feature_flag_enabled() -> bool:
+    """
+    This is temporary only and will be removed once
+    https://github.com/single-cell-data/TileDB-SOMA/issues/2407
+    is complete.
+    """
+    return _new_shape_feature_flag
```
Comment (thetorpedodog) [inline apis/python/src/tiledbsoma/_flags.py:18]: What's the motivation behind having this live in a function? I would make `NEW_SHAPE_FEATURE_FLAG_ENABLED` be a public constant and use that.

**Review by thetorpedodog — APPROVED** [review]
the move is fine; I do have one suggestion for the implementation while you’re moving things that you can take or leave.

File: apis/python/src/tiledbsoma/_flags.py (hunk 1 above)
Comment (johnkerl) [reply apis/python/src/tiledbsoma/_flags.py:18]: @thetorpedodog sure, whatever you like.

Comment (johnkerl) [reply apis/python/src/tiledbsoma/_flags.py]: https://github.com/single-cell-data/TileDB-SOMA/pull/3005/commits/a8b5909bdc0d6c037454eb389a7afbe318c92a3d

**Comment by johnkerl** [comment]
Thanks @thetorpedodog ! :)

**Pull request merged**